src/configuration.ts

```diff
@@ -124,8 +131,28 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        const configFile: IConfigFile = JSON.parse(fileData);
+        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, path.dirname(configFilePath));
+        return configFile;
+    }
+}
+
+export function extendConfigFile(config: IConfigFile, baseConfig: IConfigFile): IConfigFile {
+    let combinedConfig: IConfigFile = {};
+
+    const baseRulesDirectory = arrayify(baseConfig.rulesDirectory);
+    const configRulesDirectory = arrayify(config.rulesDirectory);
+    combinedConfig.rulesDirectory = configRulesDirectory.concat(baseRulesDirectory);
+
+    combinedConfig.rules = {};
+    for (const name of Object.keys(objectify(baseConfig.rules))) {
+        combinedConfig.rules[name] = baseConfig.rules[name];
+    }
+    for (const name of Object.keys(objectify(config.rules))) {
+        combinedConfig.rules[name] = config.rules[name];
     }
+
+    return combinedConfig;
```

Comment:
Wait:
* We don't depend on lodash AFAIK, do we want to add it?
* `_.defaults` won't concat arrays or anything, which means that things like rules directories from extended configs may not get added, which is bad.

> This is more fragile as you need to change this function every time a new field is added.

This is true, but theoretically we're not going to be adding new fields to `tslint.json` all that often...